app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,41 @@
+class ReportsController < ApplicationController
+
+  def index
+    @date = params[:date_report] ? Date.strptime(params[:date_report], "%m/%Y") : Date.today
+
+    @q = Report.where(user: current_user,
```

Comment:
Report.where(user: current_user,
+                            :date => @date.beginning_of_month..@date.end_of_month).ransack(params[:q])
+    @reports = @q.result.order('date DESC')

I think this code can be placed in model. So the controller will be drier.